### PR TITLE
Beta - Implement multi-language in giddh

### PR DESCRIPTION
**Pull request by ravinderwalkover**
https://app.clickup.com/t/1h82ed

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/ledger.component.html [hunk 1]
```diff
@@ -1,1561 +1,1433 @@
...
+                                                    <th class="col-4 col-md-3 col-lg-3 text-right">{{commonLocaleData?.app_amount}}</th>
+                                                </tr>
+                                            </thead>
+                                        </table>
+                                        <table id="ledger-transaction-left-list" class="width-100" (clickOutside)="hideBankLedgerPopup(true)" (click)="$event.stopPropagation()">
+                                            <tbody>
+                                                <ng-container *ngFor="let item of lc.bankTransactionsData; trackBy: trackByTransactionId; let i = index">
+                                                    <ng-container *ngFor="let txn of item.transactions;">
+                                                        <ng-container *ngIf="txn.type === 'DEBIT'">
+                                                            <tr class="ledger-row row" (mouseenter)="txn.showDropdown = true;" (mouseleave)="txn.showDropdown = false;" [ngClass]="{'compoundEntry': txn.id === lc.selectedBankTxnUniqueName}">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.html:251]: Please check this class name change

File: apps/web-giddh/src/app/permissions/components/list/permission-list.html [hunk 2]
```diff
@@ -1,52 +1,53 @@
...
+<ng-container appTranslate [file]="'permissions'" (localeData)="localeData = $event" (commonLocaleData)="commonLocaleData = $event">
+    <hamburger-menu [pageHeading]="localeData?.page_heading"></hamburger-menu>
+    <div class="container-fluid clearfix" *ngIf="allRoles.length > 0">
+        <a href="javascript:void(0)" *ngIf="showBackButton" class="btn btn-sky-blue pull-right mr-t1 mr-b1" [routerLink]="['/pages/settings/permission']">{{commonLocaleData?.app_back}}</a>
+        <div class="col-sm-12 mr-t15 p-0 role clearfix" *ngFor="let role of allRoles">
+            <div class="row mr-0 ml-0">
+            <div class="col-md-3 col-sm-3 col-12 p-0 role-head cp">
+                <span class="vertical-center">
+                    <h1>{{role.name}}</h1>
+                    <div class="text-center mr-t2 onHover">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/permissions/components/list/permission-list.html:10]: Please check this class name change

File: apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.html [hunk 3]
```diff
@@ -1,448 +1,432 @@
...
-                        </th>
+                            </th>
 
-                        <th *ngIf="showFieldFilter.voucherNo" class="td-search-box nowrap voucher-width"
-                            #searchInvoiceContainer (clickOutside)="clickedOutsideEvent()">
-                            <div [hidden]="showSearchInvoiceNo">
-                                <span>Voucher No</span>
-                                <i class="icon-search" (click)="toggleSearch('invoiceNumber')"></i>
-                            </div>
+                            <th *ngIf="showFieldFilter.voucherNo" class="td-search-box nowrap" style="width: 210px;"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.html:117]: Please check this class name change

File: apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.html [hunk 4]
```diff
@@ -1,451 +1,427 @@
...
+
+                </div>
             </div>
         </div>
-    </div>
-</ng-container>
+    </ng-container>
 
-<div *ngIf="(isGetSalesDetailsInProcess$ | async)">
-    <div class="no-data mr-t2 loader-main" *ngIf="showLoader">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.html:415]: Please check this change

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts [hunk 5]
```diff
@@ -103,14 +112,14 @@ export class AccountOperationsComponent implements OnInit, AfterViewInit, OnDest
...
     public discountList$: Observable<IDiscountList[]>;
-    public taxPopOverTemplate: string = `
-  <div class="popover-content">
-  <label>Tax being inherited from:</label>
-    <ul>
-    <li>@inTax.name</li>
-    </ul>
-  </div>
-  `;
+//     public taxPopOverTemplate: string = `
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts:115]: Please check this change

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html [hunk 6]
```diff
@@ -1,11 +1,12 @@
-<form name="addAccountForm" class="create-acc-form" [formGroup]="addAccountForm">
-    <div>
+<ng-container appTranslate [file]="'accounts'" (localeData)="localeData = $event" (commonLocaleData)="commonLocaleData = $event" (translationComplete)="translationComplete($event)">
+    <form name="addAccountForm" class="create-acc-form" [formGroup]="addAccountForm">
+    <div class="">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:3]: Please revert this class change

File: apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.html [hunk 7]
```diff
@@ -1,191 +1,144 @@
...
-                        <div class="form-group pull-left mr-r15 d-inline-flex relative" *ngIf="(tags$ | async)?.length">
-
-                            <i class="glyphicon glyphicon-triangle-bottom"></i>
-                            <sh-select [options]="tags$ | async" name="selectedTag" formControlName="tagName"
-                                [isFilterEnabled]="true" (selected)="onTagSelected($event)" [placeholder]="'Select Tag'"
-                                [ItemHeight]="35" [notFoundLink]="false" (noResultsClicked)="toggleTagsModal()"
-                                [showNotFoundLinkAsDefault]="false" [width]="'170px'">
-                            </sh-select>
+                        <div class="form-group relative mr-r15 datepicker-field" *ngIf="filterForm.get('selectedDateOption').value == '1'">
+                            <div class="custom-datepicker" (click)="showGiddhDatepicker($event)">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.html:48]: Please check this class name change

File: apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.html [hunk 8]
```diff
@@ -1,191 +1,144 @@
...
+                                    formControlName="selectedDateOption" [isFilterEnabled]="true"
+                                    [placeholder]="commonLocaleData?.app_select_option" [showClear]="false" [ItemHeight]="33" width="100%">
+                                </sh-select>
+                            </div>
                         </div>
-                    </div>
-                    <div class="form-group relative mr-r15 datepicker-field" *ngIf="filterForm.get('selectedDateOption').value == '1'">
-                        <div class="custom-datepicker relative" (click)="showGiddhDatepicker($event)">
-                            <i class="icon-calender calendar-addon"></i>
-                            <input type="text" name="selectedDateRange" [value]="(selectedDateRangeUi) ? selectedDateRangeUi : ''" class="giddh-datepicker date-range-picker border-radius-2" />
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.html:52]: Please check this class name change

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/audit-logs/audit-logs.component.html [hunk 9]
```diff
@@ -1,38 +1,40 @@
...
-    <div class="row">
-        <div class="col-12 col-md-12 audit-log-sidebar" [ngClass]="!isNewVersion ? 'col-lg-3' : 'col-lg-12'">
-            <!--sidebar-->
-            <audit-logs-sidebar *ngIf="!isNewVersion"></audit-logs-sidebar>
-            <audit-logs-form #auditLogFormComponent *ngIf="isNewVersion" [fromDate]="fromDate" [toDate]="toDate"></audit-logs-form>
-            <!--/sidebar-->
-        </div>
-        <div class="col-12 d-flex align-items-center">
-            <div class="mr-t15 mr-b15 custom-datepicker-wrapper" *ngIf="isNewVersion">
-                <div class="custom-datepicker" (click)="showGiddhDatepicker($event)" *ngIf="isNewVersion">
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/audit-logs/audit-logs.component.html:13]: It's not getting removed.

File: apps/web-giddh/src/app/home/components/ratio-analysis/ratio-analysis-chart.component.html [hunk 10]
```diff
@@ -1,52 +1,46 @@
...
+            <strong>{{localeData?.chart_title}}</strong>
             <div class="pull-right dashboard-filter">
                 <i class="icon-refresh mr-l1" (click)="hardRefresh()"></i>
             </div>
         </div>
 
         <div class="card-body position-relative ratio-analysis-panel-body">
             <div class="dashboard-reconnect-block" *ngIf="requestInFlight">
-                <div class="no-data mr-t2 loader-main" *ngIf="errorMessage == ''">
+                <div class="no-data mr-t2 loader-main">
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/home/components/ratio-analysis/ratio-analysis-chart.component.html:12]: yes it's fine

File: apps/web-giddh/src/app/ledger/ledger.component.html (hunk 1 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/ledger/ledger.component.html:251]: done

File: apps/web-giddh/src/app/permissions/components/list/permission-list.html (hunk 2 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/permissions/components/list/permission-list.html:10]: Done

File: apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.html (hunk 3 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.html:117]: Done

File: apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.html (hunk 4 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.html:415]: yes its fine

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts (hunk 5 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts:115]: it wasn't getting used to commented it.

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html (hunk 6 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:3]: which change?

**Review by prateek-giddh — APPROVED** [review]

**Pull request merged**